### DISCUSSION TITLE
Cannot access devices connected to the source LAN if the target LAN is the same

### DISCUSSION THREAD

**Discussion by xXConcasXx** [strongswan/strongswan#1218] [Q&A] [answered]
Hello, I have set up an IKEv2 VPN server on my home network using Strongswan on Raspberry.

I have performed portforwarding on ports 4500 and 500 UDP on my router and I have correctly set the firewall rules on my Raspberry, in fact I am able to connect.

The purpose of this configuration is to access a storage network set up on my Raspberry with openmediavault and the printer connected on the router remotely.

Currently my router configuration is as follows:

    Raspberry IP: 192.168.1.18 (DHCP static)
    Subnet Mask: 255.255.255.0
    Gateway: 192.168.1.1
    DNS: 1.1.1.1, 1.0.0.1

    IPV6 disabled

As for the management of my public IP address, being dynamic, I connected it to the work domain and pointed it to a subdomain managed on Cloudflare thanks to an ad hoc script that updates the record every minute to Cloudflare if it detects changes in my Public IP via Cloudflare REST API.

Here is my Raspberry configuration:

> /etc/ipsec.conf

    config setup
        charondebug="ike 1, knl 1, cfg 0"
        uniqueids=no
    
    conn user
        auto=add
        compress=no
        type=tunnel
        keyexchange=ikev2
        fragmentation=yes
        forceencaps=yes
        dpdaction=clear
        dpddelay=300s
        rekey=no
        left=%any
        leftid=@vpn.*****.***
        leftcert=server-cert.pem
        leftsendcert=always
        leftsubnet=0.0.0.0/0
        right=%any
        rightid=%any
        rightauth=eap-mschapv2
        rightsourceip=10.10.10.0/24
        rightdns=1.1.1.1,1.0.0.1
        rightsendcert=never
        eap_identity=%identity
        ike=chacha20poly1305-sha512-curve25519-prfsha512,aes256gcm16-sha384-prfsha384-ecp384,aes256-sha1-modp1024,aes128-sha1-modp1024,3des-sha1-modp1024!
        esp=chacha20poly1305-sha512,aes256gcm16-ecp384,aes256-sha256,aes256-sha1,3des-sha1!

> /etc/ipsec.secrets

    # This file holds shared secrets or RSA private keys for authentication.
    
    # RSA private key for this host, authenticating it to any other host
    # which knows the public part.
    
    : RSA "server-key.pem"
    user : EAP "&iC3OdvqAKRJbvSy18@IDBpvu0Kee*@S"

> /etc/ufw/before.rules

    #
    # rules.before
    #
    # Rules that should be run before the ufw command line added rules. Custom
    # rules should be added to one of these chains:
    #   ufw-before-input
    #   ufw-before-output
    #   ufw-before-forward
    #
    
    # Don't delete these required lines, otherwise there will be errors
    
    *nat
    -A POSTROUTING -s 10.10.10.0/24 -o eth0 -m policy --pol ipsec --dir out -j ACCEPT
    -A POSTROUTING -s 10.10.10.0/24 -o eth0 -j MASQUERADE
    COMMIT
    
    *mangle
    -A FORWARD --match policy --pol ipsec --dir in -s 10.10.10.0/24 -o eth0 -p tcp -m tcp --tcp-flags SYN,RST SYN -m tcpmss --mss 1361:1536 -j TCPMSS --set-mss 1360
    COMMIT
    
    *filter
    :ufw-before-input - [0:0]
    :ufw-before-output - [0:0]
    :ufw-before-forward - [0:0]
    :ufw-not-local - [0:0]
    # End required lines
    
    -A ufw-before-forward --match policy --pol ipsec --dir in --proto esp -s 10.10.10.0/24 -j ACCEPT
    -A ufw-before-forward --match policy --pol ipsec --dir out --proto esp -d 10.10.10.0/24 -j ACCEPT
    
    # allow all on loopback
    -A ufw-before-input -i lo -j ACCEPT
    -A ufw-before-output -o lo -j ACCEPT
    
    # quickly process packets for which we already have a connection
    -A ufw-before-input -m conntrack --ctstate RELATED,ESTABLISHED -j ACCEPT
    -A ufw-before-output -m conntrack --ctstate RELATED,ESTABLISHED -j ACCEPT
    -A ufw-before-forward -m conntrack --ctstate RELATED,ESTABLISHED -j ACCEPT
    
    # drop INVALID packets (logs these in loglevel medium and higher)
    -A ufw-before-input -m conntrack --ctstate INVALID -j ufw-logging-deny
    -A ufw-before-input -m conntrack --ctstate INVALID -j DROP
    
    # ok icmp codes for INPUT
    -A ufw-before-input -p icmp --icmp-type destination-unreachable -j ACCEPT
    -A ufw-before-input -p icmp --icmp-type time-exceeded -j ACCEPT
    -A ufw-before-input -p icmp --icmp-type parameter-problem -j ACCEPT
    -A ufw-before-input -p icmp --icmp-type echo-request -j ACCEPT
    
    # ok icmp code for FORWARD
    -A ufw-before-forward -p icmp --icmp-type destination-unreachable -j ACCEPT
    -A ufw-before-forward -p icmp --icmp-type time-exceeded -j ACCEPT
    -A ufw-before-forward -p icmp --icmp-type parameter-problem -j ACCEPT
    -A ufw-before-forward -p icmp --icmp-type echo-request -j ACCEPT
    
    # allow dhcp client to work
    -A ufw-before-input -p udp --sport 67 --dport 68 -j ACCEPT
    
    #
    # ufw-not-local
    #
    -A ufw-before-input -j ufw-not-local
    
    # if LOCAL, RETURN
    -A ufw-not-local -m addrtype --dst-type LOCAL -j RETURN
    
    # if MULTICAST, RETURN
    -A ufw-not-local -m addrtype --dst-type MULTICAST -j RETURN
    
    # if BROADCAST, RETURN
    -A ufw-not-local -m addrtype --dst-type BROADCAST -j RETURN
    
    # all other non-local packets are dropped
    -A ufw-not-local -m limit --limit 3/min --limit-burst 10 -j ufw-logging-deny
    -A ufw-not-local -j DROP
    
    # allow MULTICAST mDNS for service discovery (be sure the MULTICAST line above
    # is uncommented)
    -A ufw-before-input -p udp -d 224.0.0.251 --dport 5353 -j ACCEPT
    
    # allow MULTICAST UPnP for service discovery (be sure the MULTICAST line above
    # is uncommented)
    -A ufw-before-input -p udp -d 239.255.255.250 --dport 1900 -j ACCEPT
    
    # don't delete the 'COMMIT' line or these rules won't be processed
    COMMIT

> /etc/ufw/sysctl.conf

    #
    # Configuration file for setting network variables. Please note these settings
    # override /etc/sysctl.conf and /etc/sysctl.d. If you prefer to use
    # /etc/sysctl.conf, please adjust IPT_SYSCTL in /etc/default/ufw. See
    # Documentation/networking/ip-sysctl.txt in the kernel source code for more
    # information.
    #
    
    # Uncomment this to allow this host to route packets between interfaces
    #net/ipv4/ip_forward=1
    #net/ipv6/conf/default/forwarding=1
    #net/ipv6/conf/all/forwarding=1
    
    # Disable ICMP redirects. ICMP redirects are rarely used but can be used in
    # MITM (man-in-the-middle) attacks. Disabling ICMP may disrupt legitimate
    # traffic to those sites.
    
    # Ignore bogus ICMP errors
    net/ipv4/icmp_echo_ignore_broadcasts=1
    net/ipv4/icmp_ignore_bogus_error_responses=1
    net/ipv4/icmp_echo_ignore_all=0
    
    # Don't log Martian Packets (impossible addresses)
    # packets
    net/ipv4/conf/all/log_martians=0
    net/ipv4/conf/default/log_martians=0
    
    #net/ipv4/tcp_fin_timeout=30
    #net/ipv4/tcp_keepalive_intvl=1800
    
    # Uncomment this to turn off ipv6 autoconfiguration
    #net/ipv6/conf/default/autoconf=1
    #net/ipv6/conf/all/autoconf=1
    
    # Uncomment this to enable ipv6 privacy addressing
    #net/ipv6/conf/default/use_tempaddr=2
    #net/ipv6/conf/all/use_tempaddr=2
    net/ipv4/ip_forward=1
    net/ipv4/conf/all/accept_redirects=0
    net/ipv4/conf/all/send_redirects=0
    
    net/ipv4/conf/all/accept_redirects=0
    net/ipv4/conf/default/accept_redirects=0
    net/ipv4/conf/all/send_redirects=0
    net/ipv6/conf/all/accept_redirects=0
    net/ipv6/conf/default/accept_redirects=0
    
    net/ipv4/ip_no_pmtu_disc=1

But there is a problem, as long as I am connected, for example, from the mobile network of my cell phone, I can access the devices on my LAN using their local IP, the Raspberry is an example since by opening port 80 in the firewall I am able to access the admin dashboard of openmediavault from it's local IP while instead connecting from another router that uses a similar configuration to mine I can no longer access my LAN.

I would like to know where I am wrong in my configuration considering that unfortunately I am not very well versed in the world of VPNs and Strongswan.

**Comment by xXConcasXx** [comment]
I include a small outline to simplify the above

![vpn](https://user-images.githubusercontent.com/25396458/184289685-7a2e11bd-6cc5-491f-96d7-6aafdd8d2a48.png)

**Comment by Thermi** [comment, marked as answer]
This is not a strongSwan issue but an IP reuse conflict. Either renumber your internal network or use, for example, the NETMAP iptables target to setup transparent NAT for the traffic. But the better solution is to renumber your internal network to use something exotic like 192.168.83.0/24 or similar.

**Reply by xXConcasXx** [reply]
Thanks for the reply, in fact I did so while waiting to receive an answer, and everything seems to be ok.
